resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesUtils.scala

```diff
@@ -323,4 +324,15 @@ private[spark] object KubernetesUtils extends Logging {
         .build()
     }
   }
+
+  /**
+   * Convert MEMORY_OFFHEAP_SIZE to MB Unit, return 0 if MEMORY_OFFHEAP_ENABLED is false.
+   */
+  def executorOffHeapMemorySizeAsMb(kubernetesConf: KubernetesConf): Int = {
```

Comment:
we now have essentially the same code in both yarn and k8s, perhaps we can consolidate it into a utility function.